### PR TITLE
Vault benchmarking

### PR DESCRIPTION
**Pull request by hussein-aitlahcen**

### CODE REVIEW

File: frame/vault/src/lib.rs [hunk 1]
```diff
@@ -98,6 +103,97 @@ pub mod pallet {
...
+		type MinimumDeposit: Get<Self::Balance>;
+		type MinimumWithdrawal: Get<Self::Balance>;
+		type CreationDeposit: Get<Self::Balance>;
+		type ExistentialDeposit: Get<Self::Balance>;
+		type TombstoneDuration: Get<Self::BlockNumber>;
+		type RentPerBlock: Get<Self::Balance>;
+		type PalletId: Get<PalletId>;
+	}
+
+	#[cfg(not(feature = "runtime-benchmarks"))]
```
Comment (andor0) [inline frame/vault/src/lib.rs:196]: It needs for an off-chain worker only.

File: frame/vault/src/mocks/tests.rs [hunk 2]
```diff
@@ -136,6 +156,33 @@ impl crate::mocks::strategy::Config for Test {
...
+	fn create_transaction<C: frame_system::offchain::AppCrypto<Self::Public, Self::Signature>>(
+		call: Call,
+		_public: <Signature as Verify>::Signer,
+		_account: AccountId,
+		nonce: u64,
+	) -> Option<(Call, <Extrinsic as ExtrinsicT>::SignaturePayload)> {
+		Some((call, (nonce, ())))
+	}
+}
+
```
Comment (andor0) [inline frame/vault/src/mocks/tests.rs:185]: It needs for an off-chain worker only.

**Review by andor0 — CHANGES_REQUESTED** [review]

File: frame/vault/src/lib.rs (hunk 1 above)
Comment (andor0) [inline frame/vault/src/lib.rs:196]: It needs for an off-chain worker only.

**Review by andor0 — CHANGES_REQUESTED** [review]

**Comment by andor0** [comment]
imho, tests and mocks can be without changes if remove changes related to an off-chain worker.

File: frame/vault/src/lib.rs (hunk 1 above)
Comment (hussein-aitlahcen) [reply frame/vault/src/lib.rs:196]: I think i still need it as I need more constraint such as Into/From for some types...

Comment (andor0) [reply frame/vault/src/lib.rs:196]: Ok, but `crypto` module (109-139 lines) looks redundant.
`CreateSignedTransaction<Call<Self>>` needs for an off-chain worker only.

Comment (hussein-aitlahcen) [reply frame/vault/src/lib.rs:196]: I will remove all the tweaking done for the accountid of course.

**Comment by hussein-aitlahcen** [comment]
@andor0 fixed, any idea on how to make cargofmt pass? I'm using latest rustfmt and it's clashing with the expectation here :thinking: (rustfmt 1.4.38-nightly (baba668 2021-11-03))

**Comment by andor0** [comment]
> @andor0 fixed, any idea on how to make rustfmt pass? I'm using latest rustfmt and it's clashing with the expectation here thinking (rustfmt 1.4.38-nightly (baba668 2021-11-03))

Updated CI config.

File: frame/vault/src/lib.rs [hunk 3]
```diff
@@ -98,6 +105,99 @@ pub mod pallet {
...
+			frame_system::offchain::AppCrypto<
+				<Sr25519Signature as Verify>::Signer,
+				Sr25519Signature,
+			> for TestAuthId
+		{
+			type RuntimeAppPublic = Public;
+			type GenericSignature = sp_core::sr25519::Signature;
+			type GenericPublic = sp_core::sr25519::Public;
+		}
+	}
```
Comment (andor0) [inline frame/vault/src/lib.rs:141]: compilation succeed without this code

Comment (hussein-aitlahcen) [reply frame/vault/src/lib.rs]: Thats my bad I forget it.

**Review by andor0 — APPROVED** [review]

**Pull request merged**